tomlparser/toml.go

```diff
@@ -0,0 +1,315 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package tomlparser
+
+// TODO: Currently toml parser gets a single diagnostic at a time. After migrating the parser, should use the same for toml as well.
+
+import (
+	"errors"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+type Toml struct {
+	rootNode    map[string]any
+	metadata    toml.MetaData
+	diagnostics []Diagnostic
+	content     string
+}
+
+type Diagnostic struct {
+	Message  string
+	Severity string
+	Location *Location
+}
+
+type Location struct {
+	StartLine   int
+	StartColumn int
+	EndLine     int
+	EndColumn   int
+}
+
+func readFile(path string) (string, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+func readFromReader(reader io.Reader) (string, error) {
+	content, err := io.ReadAll(reader)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+func Read(path string) (*Toml, error) {
```

Comment:
I think for cases we accept paths we should accept [FileSystem](https://pkg.go.dev/io/fs#pkg-overview) as well. Then instead of directly using io, which will cause problems downstream when we try to run this in the browser we can write these agnostic towards the actual file system. (For browser we can write (if there is none) our own file system)